hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AEndpointRegion.java

```diff
@@ -350,32 +353,33 @@ public void testCentralEndpointAndDifferentRegionThanBucket() throws Throwable {
   @Test
   public void testWithOutCrossRegionAccess() throws Exception {
     describe("Verify cross region access fails when disabled");
+    // skip the test if the region is sa-east-1
+    skipCrossRegionTest();
     final Configuration newConf = new Configuration(getConfiguration());
-    // skip the test if the region is eu-west-2
-    String region = getFileSystem().getS3AInternals().getBucketMetadata().bucketRegion();
-    if (EU_WEST_2.equals(region)) {
-      return;
-    }
     // disable cross region access
     newConf.setBoolean(AWS_S3_CROSS_REGION_ACCESS_ENABLED, false);
-    newConf.set(AWS_REGION, EU_WEST_2);
-    S3AFileSystem fs = new S3AFileSystem();
-    fs.initialize(getFileSystem().getUri(), newConf);
-    intercept(AWSRedirectException.class,
-        "does not match the AWS region containing the bucket",
-        () -> fs.exists(getFileSystem().getWorkingDirectory()));
+    newConf.set(AWS_REGION, SA_EAST_1);
+    try (S3AFileSystem fs = new S3AFileSystem()) {
+      fs.initialize(getFileSystem().getUri(), newConf);
+      intercept(AWSRedirectException.class,
+          "does not match the AWS region containing the bucket",
+          () -> fs.exists(getFileSystem().getWorkingDirectory()));
+    }
   }
 
   @Test
   public void testWithCrossRegionAccess() throws Exception {
     describe("Verify cross region access succeed when enabled");
+    // skip the test if the region is sa-east-1
+    skipCrossRegionTest();
     final Configuration newConf = new Configuration(getConfiguration());
```

Comment:
call
```
removeBaseAndBucketOverrides(newConf,
   AWS_S3_CROSS_REGION_ACCESS_ENABLE,
   AWS_REGION)
```

needed to strip out per bucket settings of these options.

Reply:
ack